internals/controllers/migrationhierarchy_controller.go

```diff
@@ -137,10 +138,15 @@ func (r *MigrationHierarchyReconciler) Reconcile(ctx context.Context, req ctrl.R
 			return object, log
 		})
 
+		if err := namespaceDB.MigrateNsHierarchy(r.NamespaceDB, r.Client, r.Log, ns.GetName(), toNs.GetName()); err != nil {
```

Comment:
Add a short comment explaining that we need to update the DB to account for the new parent